x-pack/plugins/fleet/server/services/agents/crud.ts

```diff
@@ -415,6 +415,46 @@ export async function getAgentById(
   return agentHit;
 }
 
+/**
+ * Get list of agents by `id`. service method performs space awareness checks.
+ * @param esClient
+ * @param soClient
+ * @param agentIds
+ * @param options
+ *
+ * @throws AgentNotFoundError
+ */
+export const getByIds = async (
```

Comment:
Also perhaps rename this to `getAgentsByIds`

Reply:
same comment as before - just following Fleet conventions as other `getByIds()` already exist